lib/engine-addon.js

```diff
@@ -297,7 +297,12 @@ var buildCompleteJSTree = memoize(function buildCompleteJSTree() {
 function maybeMergeTrees(_inputTrees, options) {
   var inputTrees = _inputTrees.filter(Boolean);
   if (inputTrees.length > 1) {
-    return mergeTrees(inputTrees, options);
+    let merged = mergeTrees(inputTrees, options);
+    if (Array.isArray(merged._inputTrees) && merged._inputTrees.length === 0) {
```

Comment:
This is odd, how can we ever get here?

Reply:
mergeTrees is the special mergeTrees you wrote, which is EMPTY_TREE aware.

Reply:
@rwjblue any idea of test this function/flow? I don't can export this function to test only.